lib/Checks/MissingBracketsCheck.vala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016 elementary LLC. (https://github.com/elementary/Vala-Lint)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Authored by: Sören Busse
+ */
+
+public class ValaLint.Checks.MissingBracketsCheck : Check {
+    private Regex regex_no_brackets;
+    private Regex regex_same_line;
+    private Regex regex_same_line_no_brackets;
+
+    public MissingBracketsCheck () {
+        try {
+            regex_no_brackets = new Regex ("^(if|while|for|foreach) *\\(.*\\) *$");
+            regex_same_line = new Regex ("(if|while|for|foreach) *\\(.*\\) *[{]+ *[a-zA-Z0-9_()]+");
+            regex_same_line_no_brackets = new Regex ("(if|while|for|foreach) *\\(.*\\) *[^{] *[a-zA-Z0-9_()]+");
+        } catch (Error e) {
+            /* Should never happen */
+            error ("There was a parsing error during doing some regex");
```

Comment:
Maybe print the error message here, too:
`error ("There was a parsing error during doing some regex: %s", e.message);`
And no need for comments like "should never happen". We know that especially these things *do* happen. 😂